### DISCUSSION TITLE
next/image should not force async decoding, as it can slow down the overall page load and cause flashes

### DISCUSSION THREAD

**Discussion by cristianlivella** [vercel/next.js#50524] [Ideas]
### Goals

1. allow developers to have more control over image loading optimization
2. make smoother websites

### Background

I'm not actually sure about why `decoding="async"` is forced for next/image, but I think it should be possibile to override it, because in some cases it isn't the best choice, as it slow down the overall page load and the user can perceives flashes due to the images being shown later than the other content.

For example, check [exallievi-new.itispaleocapa.it](https://exallievi-new.itispaleocapa.it/), and you can see the effect I'm talking about. You can even easily spot it with PageSpeed Insights:
![image](https://github.com/vercel/next.js/assets/27968888/c1674f87-d7fe-4cd2-b977-14d03baad07d)

Here you can see the same website, but without `decoding="async"` on the images: [exallievi.itispaleocapa.it](https://exallievi.itispaleocapa.it/).
![image](https://github.com/vercel/next.js/assets/27968888/e096db3a-561c-4005-beea-6dd08e6bd108)

### Proposal

[Here](https://github.com/vercel/next.js/blob/7adb273b02404949784f9211438e1d5d32e3b6ee/packages/next/src/client/image.tsx#LL427C9-L427C25) move the line `decoding="async"` before `{...rest}`.

**Comment by quarxpl** [comment]
We are encountering a similar issue, which appears to be specific to Chromium-based browsers. It primarily affects large images, which load at a considerably slow pace when decoding is set to async. Small images seem unaffected. Our current makeshift solution is a reimplementation of the <Image> component, used selectively for some images. This, however, is far from being an ideal solution.

**Reply by adardesign** [reply]
Hi, any update on this?
https://github.com/vercel/next.js/pull/49772

**Comment by UPSxACE** [comment]
Workaround I am using for now:
https://github.com/vercel/next.js/pull/49772#issuecomment-2132166453
